pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -351,19 +346,6 @@ func (s *Server) makeKubeConfigController(args *PilotArgs) *crdclient.Client {
 	return crdclient.New(s.kubeClient, opts)
 }
 
-func (s *Server) makeFileMonitor(fileDir string, domainSuffix string, configController model.ConfigStore) error {
-	fileSnapshot := configmonitor.NewFileSnapshot(fileDir, collections.Pilot, domainSuffix)
-	fileMonitor := configmonitor.NewMonitor("file-monitor", configController, fileSnapshot.ReadConfigFiles, fileDir)
-
-	// Defer starting the file monitor until after the service is created.
-	s.addStartFunc("file monitor", func(stop <-chan struct{}) error {
-		fileMonitor.Start(stop)
```

Comment:
previously this controller would only start monitoring files when the server actually started, we are losing this with the current implementation, the KRT implementation does not allow for a delayed start, but we could delay the creation of collections if it's necessary.